mangaki/mangaki/views.py

```diff
@@ -749,6 +747,37 @@ def update_newsletter(request):
     return HttpResponse()
 
 
+def update_research(request):
+    is_ok = None
+    if request.user.is_authenticated and request.method == 'POST' and 'research_ok' in request.POST:  # Toggle on one's profile
+        username = request.user.username
+        is_ok = request.POST.get('research_ok') == 'true'
+        print(is_ok, 'wwahahaa')
+        Profile.objects.filter(user__username=username).update(research_ok=is_ok)
+        return HttpResponse()
+    if request.method == 'POST':  # Confirmed from mail link
+        is_ok = 'yes' in request.POST
```

Comment:
Because the user chooses the submit button on which they want to click.

    <input type="submit" class="btn btn-danger" name="no" value="Ne pas participer" />
    <input type="submit" class="btn btn-success" name="yes" value="Participer" />